File: clothing-management-frontend/src/app/services/packet.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Packet } from 'src/shared/models/Packet';

@Injectable({
  providedIn: 'root'
})
export class PacketService {

  private baseUrl: string = "http://localhost:2233/packet";

  constructor(private http: HttpClient) { }

  findAllPackets() {
    return this.http.get(this.baseUrl + "/findAll");
  }

  findPacketById(id: number) {
    return this.http.get(this.baseUrl + "/findById/" + id);
  }
  
  findPacketRelatedProducts(id: number) {
    return this.http.get(this.baseUrl + "/findPacketRelatedProducts/" + id);
  }

  addPacket(packet: Packet) {
    return this.http.post(this.baseUrl + "/add" , packet , {observe: 'body'})
  }

  updatePacket(packet: Packet) {
    return this.http.put(this.baseUrl + "/update" , packet , {headers : { 'content-type': 'application/json'}})
  }

  patchPacket(idPacket: any , packet: any) {
    console.log(packet);
    return this.http.patch(this.baseUrl + "/patch/" + idPacket , packet , {headers : { 'content-type': 'application/json'}})
  }

  addProductsToPacket(selectedProducts: any) {
    console.log(selectedProducts);
    return this.http.post(this.baseUrl + "/addProducts" , selectedProducts , {headers : { 'content-type': 'application/json'}})
  }

  deletePacketById(idPacket: any) {
    console.log(this.baseUrl + "/deleteById/" + idPacket)
    return this.http.delete(this.baseUrl + "/deleteById/" + idPacket)
  }

  deleteSelectedPackets(packetsId: any[]) {
    return this.http.delete(this.baseUrl + "/deleteSelectedPackets/" + packetsId);
  }

}

File: clothing-management-backend/src/main/java/com/clothing/management/services/PacketService.java
package com.clothing.management.services;

import com.clothing.management.dto.OfferUpdateDTO;
import com.clothing.management.dto.SelectedProductsDTO;
import com.clothing.management.entities.Packet;
import org.springframework.web.bind.annotation.PathVariable;

import java.util.Date;
import java.util.List;
import java.util.Map;
import java.util.Optional;

public interface PacketService {

    public List<Packet> findAllPackets();
    public List<Packet> findAllTodaysPackets();
    public List<Packet> findAllPacketsByDate(Date date);
    public Optional<Packet> findPacketById(Long idPacket);
    public List<OfferUpdateDTO> findPacketRelatedProducts(Long idPacket);
    public Packet addPacket(Packet packet);
    public Packet updatePacket(Packet packet);
    public Packet patchPacket(Long idPacket , Map<String , Object> packet);
    public void addProductsToPacket(SelectedProductsDTO selectedProductsDTO);
    public void deletePacketById(Long idPacket);
    public void deleteSelectedPackets(List<Long> packetsId);
}
